specification/keyvault/Security.KeyVault.Certificates/models.tsp

```diff
@@ -0,0 +1,1043 @@
+}
+
+/**
+ * The type of key pair to be used for the certificate.
+ */
+union JsonWebKeyType {
+  string,
+
+  /** Elliptic Curve. */
+  EC: "EC",
+
+  /** Elliptic Curve with a private key which is not exportable from the HSM. */
+  EC_HSM: "EC-HSM",
+
+  /** RSA (https://tools.ietf.org/html/rfc3447). */
+  RSA: "RSA",
+
+  /** RSA with a private key which is not exportable from the HSM. */
+  RSA_HSM: "RSA-HSM",
+
+  /** Octet sequence (used to represent symmetric keys). */
+  oct: "oct",
+
+  /** Octet sequence with a private key which is not exportable from the HSM. */
+  oct_HSM: "oct-HSM",
+}
+
+/**
+ * Elliptic curve name. For valid values, see JsonWebKeyCurveName.
+ */
+union JsonWebKeyCurveName {
+  string,
+
+  /** The NIST P-256 elliptic curve, AKA SECG curve SECP256R1. */
+  P_256: "P-256",
+
+  /** The NIST P-384 elliptic curve, AKA SECG curve SECP384R1. */
+  P_384: "P-384",
+
+  /** The NIST P-521 elliptic curve, AKA SECG curve SECP521R1. */
+  P_521: "P-521",
+
+  /** The SECG SECP256K1 elliptic curve. */
+  P_256K: "P-256K",
+}
+
+/**
+ * Supported usages of a certificate key.
+ */
+union KeyUsageType {
+  string,
+
+  /** Indicates that the certificate key can be used as a digital signature. */
+  digitalSignature: "digitalSignature",
+
+  /** Indicates that the certificate key can be used for authentication. */
+  nonRepudiation: "nonRepudiation",
+
+  /** Indicates that the certificate key can be used for key encryption. */
+  keyEncipherment: "keyEncipherment",
+
+  /** Indicates that the certificate key can be used for data encryption. */
+  dataEncipherment: "dataEncipherment",
+
+  /** Indicates that the certificate key can be used to determine key agreement, such as a key created using the Diffie-Hellman key agreement algorithm. */
+  keyAgreement: "keyAgreement",
+
+  /** Indicates that the certificate key can be used to sign certificates. */
+  keyCertSign: "keyCertSign",
+
+  /** Indicates that the certificate key can be used to sign a certificate revocation list. */
+  cRLSign: "cRLSign",
+
+  /** Indicates that the certificate key can be used for encryption only. */
+  encipherOnly: "encipherOnly",
+
+  /** Indicates that the certificate key can be used for decryption only. */
+  decipherOnly: "decipherOnly",
+}
+
+/**
+ * The type of the action.
+ */
+#suppress "@azure-tools/typespec-azure-core/no-enum" "Intentional enum; originally modelAsString=false"
+enum CertificatePolicyAction {
+  /** A certificate policy that will email certificate contacts. */
+  EmailContacts: "EmailContacts",
+
+  /** A certificate policy that will auto-renew a certificate. */
+  AutoRenew: "AutoRenew",
+}
+
+/**
+ * The certificate list result.
+ */
+@pagedResult
+model CertificateListResult {
+  /**
+   * A response message containing a list of certificates in the key vault along with a link to the next page of certificates.
+   */
+  @items
+  @visibility("read")
```

Comment:
You should not need the visibility read here since this is presumably only used in a response?

Reply:
I tried removing this visibility decorator but then the compiled Swagger loses the `"readOnly": true` attribute on the items, unfortunately.

Reply:
readOnly: true is nonsensical for a response-only schema though. So that is a difference that shouldn't matter. And if @tjprescott 's diff tool says it does, then we should talk to him about it.

Reply:
Fair -- our main concern with matching the Swagger is to avoid breaking docs generated from it, but [CertificateListResult](https://learn.microsoft.com/en-us/rest/api/keyvault/certificates/get-certificate-versions/get-certificate-versions?view=rest-keyvault-certificates-7.4&tabs=HTTP#certificatelistresult)'s documentation makes me think it would be fine to lose the `readOnly` here. @heaths, thoughts?

Reply:
While my thinking is aligned with yours, if @johanste is saying it's okay and @tjprescott's differ doesn't flag it I don't see a problem dropping `readOnly`, no. If the differ does flag it, though, we can file a tracking issue once the differ gets updated. We can't wait on merging this much longer over seemingly trivial matters.

Reply:
Looks like it worked. @johanste was right.